### DISCUSSION TITLE
Support nested switch statements in DataObjFormatters

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5666] [Feature Requests]
Requested by CSIRO:

> We currently have our format for the determination table differ based on the isCurrent field (see image) but I was wondering if there was a way to check multiple fields? In our fish collection we have the isCurrent field but also another boolean field called "Label". We'd like to have four different table formats:
isCurrent=true & Label=true | prefferedTaxon.fullName (current) (label)
isCurrent=true & Label=false| prefferedTaxon.fullName (current)
isCurrent=false & Label=true | prefferedTaxon.fullName (label)
isCurrent=false & Label=false | prefferedTaxon.fullName

> Is this possible? I can see that the UI in Specify6 only lets us pick one dependent field but I can see that the code is written into the DataObjFormatters app resource - can I change the code here to achieve the example above? Our current code is below for reference.

```xml
<format
        name="Determination"
        title="Determination"
        class="edu.ku.brc.specify.datamodel.Determination"
        >
        <switch single="false" field="isCurrent">
            <fields value="true">
                <field>preferredTaxon.fullName</field>
                <field type="boolean" format=" " sep=" (current)">isCurrent</field>
                <field type="boolean" sep=" - (label)">yesNo1</field>
            </fields>
            <fields value="false">
                <field>taxon.fullName</field>
            </fields>
        </switch>
    </format>
```

**Comment by grantfitzsimmons** [comment]
This would be nice.

**Comment by maxpatiiuk** [comment]
Optional TODO here:
Check if sp6 includes support for this in the code (but not in the UI)
Because I remember Ben opened an issue at one point saying to add this feature (at least 5 years ago), which gives me an impression this might be implemented behind the scenes in sp6

**Comment by maxpatiiuk** [comment]
FYI: there are workarounds for this feature in many cases, especially when a -one-to-one is available:
for table A have a formatter with the first switch, and it would call an appropriate formatter for a -one-to-one table B
Table B formatter has the 2nd switch, and then displays the fields from table B or though relationship back from table A as desired.

With -to-manys it's a bit trickier, but still sometimes possible:
have an aggregator that calls a formatter, that has a switch that displays nothing if some condition is not met (i.e, determination is not current) - this way only a single record in a -to-many will be displayed.
Alternatively, on the aggregator definition, you can say to sort fields by a field, and display only the first result
